src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -86,26 +83,57 @@ struct LocationStructureWrapper : public chip::app::Clusters::ServiceArea::Struc
      *
      * @note If the locationName is empty string and aFloorNumber and aAreaTypeTag are null, locationInfo will be set to null.
      */
-    LocationStructureWrapper & operator=(const LocationStructureWrapper & aOther)
+    AreaStructureWrapper & operator=(const AreaStructureWrapper & aOther)
     {
-        if (aOther.locationInfo.locationInfo.IsNull())
+        areaID = aOther.areaID;
+        mapID  = aOther.mapID;
+
+        if (aOther.areaDesc.locationInfo.IsNull())
+        {
+            areaDesc.locationInfo.SetNull();
+        }
+        else
+        {
+            areaDesc.locationInfo.SetNonNull();
+
+            // deep copy the name.
+            auto sizeToCopy = sizeof(mLocationNameBuffer);
+            if (aOther.areaDesc.locationInfo.Value().locationName.size() < sizeToCopy)
+            {
+                sizeToCopy = aOther.areaDesc.locationInfo.Value().locationName.size();
+            }
+            memcpy(mLocationNameBuffer, aOther.areaDesc.locationInfo.Value().locationName.data(), sizeToCopy);
```

Comment:
Assuming this is allowed in the server code + by the style guide, consider using an alias for aOther.areaDesc.locationInfo.Value() (a ref to it), to avoid copying that long "path" in the implementation code.

Reply:
Leaving as is as it is more explicit.